lib/utils.c

```diff
@@ -39,6 +39,46 @@
 
 int timestamp_short;
 
+int read_prop(char *dev, char *prop, long *value)
+{
+	char fname[128], buf[80], *endp;
+	FILE *fp;
+	long result;
+	int ret;
+
+	ret = snprintf(fname, sizeof(fname), "/sys/class/net/%s/%s", dev, prop);
+	if (ret <= 0 || ret >= sizeof(fname))
+                return -1;
```

Comment:
Let's add an error message here: `fprintf(stderr, "could not build pathname for property\n");`